airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -112,22 +117,170 @@ def test_connection(self) -> tuple[bool, str]:
             self.log.error("Error testing Weaviate connection: %s", e)
             return False, str(e)
 
+    @retry(reraise=True, stop=stop_after_attempt(3), retry=retry_if_exception_type(requests.ConnectionError))
     def create_class(self, class_json: dict[str, Any]) -> None:
         """Create a new class."""
         client = self.get_client()
         client.schema.create_class(class_json)
 
-    def create_schema(self, schema_json: dict[str, Any]) -> None:
+    @retry(reraise=True, stop=stop_after_attempt(3), retry=retry_if_exception_type(requests.ConnectionError))
+    def create_schema(self, schema_json: dict[str, Any] | str) -> None:
         """
         Create a new Schema.
 
         Instead of adding classes one by one , you can upload a full schema in JSON format at once.
 
-        :param schema_json: The schema to create
+        :param schema_json: The schema to create or path to the json file holding the schema
         """
         client = self.get_client()
         client.schema.create(schema_json)
 
+    @retry(reraise=True, stop=stop_after_attempt(3), retry=retry_if_exception_type(requests.ConnectionError))
+    def get_schema(self, class_name: str | None = None):
+        """Get the schema from Weaviate.
+
+        get schema of a class or all classes.
```

Comment:
what does all classes mean here?

Reply:
Ya, it's a little confusing, but yes the doc string can be improved. 
So we can think of schema as a list and class as an item within that list. With this method, we can get config for all the classes within a schema or a specific class.

Reply:
Updated the docstring.